crypto/asn1/a_dup.c

```diff
@@ -64,5 +69,12 @@ void *ASN1_item_dup(const ASN1_ITEM *it, const void *x)
     p = b;
     ret = ASN1_item_d2i(NULL, &p, i, it);
     OPENSSL_free(b);
+    if (asn1_cb != NULL
+            && !asn1_cb(ASN1_OP_DUP_POST, (ASN1_VALUE **)&ret, it, (void *)x))
```

Comment:
If you change ret to be `ASN1_VALUE *`, you won't need this cast...

Reply:
Good point - done.